### PR TITLE
first approach

### PR DESCRIPTION
**Pull request by howls90**

### CODE REVIEW

File: apis/wordcount.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+@api.route('/')
+class Count(Resource):
+    @api.doc('WordCount')
+    @api.doc(parser=count_request)
+    @api.marshal_with(count_model, code=200)
+    @api.doc(responses={
+        500: 'Internal Server Error',
+        400: 'Validation Error'
+    })
```
Comment (howls90) [inline apis/wordcount.py]: To make the description of the API, I'm using swagger. I can make the description of the API code inside the code or outside in a different folder. Some developers prefer inside, others outside. Do you have any recommendation or is up to me??

File: apis/wordcount.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        '''Count number of times word appear in URL'''
+        request = WordCountRequest(count_request.parse_args())
+
+        count = db.read(request.url)
+        if count is not None:
+            request.count = int(count)
+        else:        
+            data = pullOutData(request.url)
+            request.count = findNumberTimes(data, request.word)
+            db.save(request.url, request.count)
```
Comment (howls90) [inline apis/wordcount.py]: I'm putting the logic inside the "controller" but some people prefer use services to not merge endpoints and logic. I prefer follow the rules of the team when I making the design of the APIs, so do you have any recommendation or is up to me?

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  # web:
+  #   image: nginx
+  #   restart: always
+  #   ports:
+  #     - "80:80"
+  #   links:
+  #     - app
+  #   volumes:
+  #     - .:/myapp
+  #     - ./nginx:/etc/nginx/conf.d
```
Comment (howls90) [inline docker-compose.yml]: The deployment of the API will be in a server or inside a lambda function? Depends on that I will build a server or not and I will do changes on the code.

File: apis/wordcount.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    @api.doc('WordCount')
+    @api.doc(parser=count_request)
+    @api.marshal_with(count_model, code=200)
+    @api.doc(responses={
+        500: 'Internal Server Error',
+        400: 'Validation Error'
+    })
+    def post(self):
+        '''Count number of times word appear in URL'''
+        request = WordCountRequest(count_request.parse_args())
```
Comment (howls90) [inline apis/wordcount.py]: I used form-data for the post request but I think is more suitable to use json if the API will be call from SPA via AJAX. Any suggestion or up to me?

File: core.py [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+    
+def pullOutData(url):
+    '''Pull out data from the url'''
+    html = urlopen(url)
+    soup = BeautifulSoup(html, 'html.parser')
+    return soup
+
+def findNumberTimes(data, word):
+    '''Parse data'''
+    num = data.find_all(text=lambda text: text and " {} ".format(word) in text) #check mes de un per line
```
Comment (howls90) [inline core.py]: I just need to count the word in the HTML inner? If the word is inside a ```<h1>``` tag or ```<p>``` doesn't matter?

File: apis/wordcount.py [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+count_model = api.model('Count', {
+    'url': fields.String(required=True, description='URL to parser'),
+    'word': fields.String(required=True, description='Word to Count'),
+    'count': fields.Integer(required=True, description='Number of times word in URL'),
+})
+
+count_request = api.parser()
+count_request.add_argument('word', 
+                            required=True,
+                            type=inputs.regex('^[a-zA-Z0-9]+$'), #check
```
[the remark this answers is not in the document]
Comment (zr0z) [reply apis/wordcount.py]: Just alphanumeric.
Case handling could be solved by a parameter, but as long as you comment the exact behavior in the README, we are fine with any choices.

File: apis/wordcount.py [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+                            help='Must be String', 
+                            location='form')
+count_request.add_argument('url', 
+                            type=inputs.URL(schemes=['http', 'https'], check=True), # allow IPS
```
[the remark this answers is not in the document]
Comment (zr0z) [reply apis/wordcount.py]: Only in the path, yes.

File: apis/wordcount.py (hunk 1 above)
Comment (zr0z) [reply apis/wordcount.py]: I prefer inside the code, but outside is fine as well.

File: apis/wordcount.py (hunk 2 above)
Comment (zr0z) [reply apis/wordcount.py]: We usually use services / managers.

File: docker-compose.yml (hunk 3 above)
Comment (zr0z) [reply docker-compose.yml]: We deploy more and more on lambdas.

File: apis/wordcount.py (hunk 4 above)
Comment (zr0z) [reply apis/wordcount.py]: JSON please! 
Both for the request and responses (valid, 404, 500)

File: core.py (hunk 5 above)
Comment (zr0z) [reply core.py]: Yes, are not filtering by tags.

**Pull request merged**